readthedocs/core/utils/url.py

```diff
@@ -0,0 +1,15 @@
+"""URL hadling utilities."""
+
+
+def join_url_path(base, *args):
+    """
+    Joins a base URL path with one or more path components.
+
+    This does a simple join of the base path with the path components,
+    inserting a slash between each component.
+    It does not offer protection against directory traversal attacks.
```

Comment:
This sounds scary. Should it?

Reply:
We aren't using this to serve files or similar, so this isn't dangerous, a path traversal here doesn't affect the redirect either, so not sure if we should raise a warning here, if anything we can check for `..` in the middleware to reject all suspicious paths.

Reply:
Yea, I think we should either:

* Update the comment about the security issue, explaining _why_ we're not checking for it here, not just noting it isn't checked for and leaving it up to the reader to know what that means
* Make the code more defensive and raise warnings (or exceptions), even if it doesn't matter, if it's not something we think should be done

I'd lean more towards having the code be extra defensive, just because that's going to help us not make silly errors in the future reusing this in a security-critical context.